### PR TITLE
Allow ammunition to burn and make small explosions with loud sounds

### LINKED ISSUE #25031: Bring back ammunition explosions as a special effect.
**Is your feature request related to a problem? Please describe.**  
#24862 addressed the most pressing issue in #24788 of lethal exploding small arms ammunition, but removed the special effect of exploding ammunition, which aids imersiveness and provides an opportunity for manipulating the environment (such as by causing a distraction).

**Describe the solution you'd like**  
Re-add the explosion, but with a very minimal magnitude, similar to a firecracker (especially, with no shrapnel).

**Describe alternatives you've considered**  
Instead of exploding, it could simply cause a noise and destroy the ammunition, but since that's exactly what the explosion code does when properly tuned, it seems unnecessary.

### PR DESCRIPTION
**Pull request by SeventhSandwich**
```SUMMARY: Feature "Allow ammunition to burn and create small explosions with loud sounds"```

#### Purpose of change
Closes #25031
Allows bullets to be quickly dropped and lit ablaze to function like a firecracker and distract zombies with noise. 

#### Describe the solution
This PR substantially decreases the rate at which a pile of ammo burns, creating an effect where a pile of ammo will burn like a string of firecrackers (with similar sound levels, as well). 

The noise scales based on quantity, but plateaus out at ~30-40 noise even with many bullets. A pile of 25 rounds will make sounds ranging from 10-20 noise that go for about 8 turns based on my testing. 

As suggested in the issue, the strength of the explosions is comparable to a firecracker for most piles of ammo greater than 10.  The factor for blast radius is set to zero. 

#### Describe alternatives you've considered
I explored the idea of making large piles of ammo have some explosive blast. Realistically speaking, you wouldn't get an explosive blast from burning ammo stores. 

source: https://www.youtube.com/watch?v=R-yqVAK9uxo

#### Testing
I built CDDA in Visual Studio 2019 and tested burning different piles of ammunition. The new effect seems well balanced in my opinion. I went through several iterations of the explosion parameters. 

#### Additional context
None.

### CODE REVIEW

File: src/item.cpp [hunk 1]
```diff
@@ -8086,6 +8086,10 @@ bool item::detonate( const tripoint &p, std::vector<item> &drops )
             // If it has a special effect just trigger it.
             apply_ammo_effects( p, ammo_type.ammo_effects );
         }
+        if( ammo_type.cookoff ) {
+            // If ammo type can burn, then create an explosion proportional to quantity.
+            explosion_handler::explosion( p, 3.0f*sqrtf(sqrtf(rounds_exploded / 25.0f)), 0.0f, false, 0 );
```
Comment (Night-Pryanik) [inline src/item.cpp]:
```suggestion
            explosion_handler::explosion( p, 3.0f*sqrtf( sqrtf( rounds_exploded / 25.0f ) ), 0.0f, false, 0 );
```

Comment (SeventhSandwich) [reply src/item.cpp]: My bad - I knew about the rules for spacing but missed that one.

Comment (anothersimulacrum) [reply src/item.cpp]: There's an automated program called astyle that we use for code format. I suggest you use that, instead of doing it manually.

Comment (SeventhSandwich) [reply src/item.cpp]: I'll check it out

File: src/item.cpp [hunk 2]
```diff
@@ -8077,7 +8077,7 @@ bool item::detonate( const tripoint &p, std::vector<item> &drops )
         return true;
     } else if( type->ammo && ( type->ammo->special_cookoff || type->ammo->cookoff ) ) {
         int charges_remaining = charges;
-        const int rounds_exploded = rng( 1, charges_remaining );
+        const int rounds_exploded = rng( 1, charges_remaining/2 );
```
Comment (kevingranade) [inline src/item.cpp:8080]:
```suggestion
        const int rounds_exploded = rng( 1, charges_remaining / 2 );
```

File: src/item.cpp [hunk 3]
```diff
@@ -8086,6 +8086,10 @@ bool item::detonate( const tripoint &p, std::vector<item> &drops )
             // If it has a special effect just trigger it.
             apply_ammo_effects( p, ammo_type.ammo_effects );
         }
+        if( ammo_type.cookoff ) {
+            // If ammo type can burn, then create an explosion proportional to quantity.
+            explosion_handler::explosion( p, 3.0f*sqrtf( sqrtf( rounds_exploded / 25.0f ) ), 0.0f, false, 0 );
```
Comment (kevingranade) [inline src/item.cpp:8091]:
```suggestion
            explosion_handler::explosion( p, 3.0f * sqrtf( sqrtf( rounds_exploded / 25.0f ) ), 0.0f, false, 0 );
```

**Pull request merged**